crates/pevm/src/lib.rs

```diff
@@ -4,14 +4,49 @@
 
 // TODO: Better types & API for third-party integration
 
-use std::hash::{BuildHasher, BuildHasherDefault, Hash, Hasher};
+use std::{
+    future::Future,
+    hash::{BuildHasher, BuildHasherDefault, Hash, Hasher},
+    sync::OnceLock,
+    thread,
+};
+use tokio::runtime::{Handle, Runtime};
 
 use alloy_primitives::{Address, B256, U256};
 use bitflags::bitflags;
 use hashbrown::HashMap;
 use rustc_hash::FxBuildHasher;
 use smallvec::SmallVec;
 
+static GLOBAL_RUNTIME: OnceLock<Runtime> = OnceLock::new();
+
+#[inline(always)]
+/// Awaits on a future using the current Tokio runtime if it exists,
+/// or creates a new one (at most once) if it doesn't.
+pub fn block_on<F: Future + Send>(future: F) -> F::Output
+where
+    F::Output: Send,
+{
+    // Get a handle to the current Tokio runtime
+    if Handle::try_current().is_ok() {
```

Comment:
This implementation seems off as it doesn't reuse the existing handle and duplicates the usage of `GLOBAL_RUNTIME` in both branches. I think the intended logic must look somewhat like this:

```rust
if let Ok(handle) = Handle::try_current() {
    handle
} else {
    GLOBAL_RUNTIME
        .get_or_init(|| Runtime::new().expect("Failed to create Tokio runtime"))
        .handle()
        .clone()
}
.block_on(future)
```

Reply:
It was but it was blocking, due to `tokio::test` using a single thread. I'll update `block_on` now that it works